packages/zoe/src/contracts/auction/index.js

```diff
@@ -116,15 +116,33 @@ const start = zcf => {
     });
   };
 
+  const BidProposalShape = M.splitRecord({
```

Comment:
I like `makeNatAmountShape()` from `packages/inter-protocol/src/contractSupport.js`.

Reply:
From a brief skim of the code, I did not figure out what expression to write in this location for the brand. ELI5, could you show me plausible code? Thanks.

Reply:
The brands aren't known until the sell invitation is exercised, so it is correct for `SellProposalShape` to stay generic. After that, we can create the bid shape:
```
    const bidBrand = sellSeat.getProposal().want.Ask.brand;
    const AmountShape = makeNatAmountShape(bidBrand);

    const BidProposalShape = M.splitRecord({
      give: { Bid: AmountShape },
      want: { Asset: AmountShape },
    });
```
`BidProposalShape` is then used in `makeBidInvitation`. We can either use `let` to set a global, or use currying to build the `makeBidInvitation` that is returned to the seller. I think I prefer the latter, so the first line of `makeBidInvitation` changes to 
```
  const buildBidInvitationMaker = bidProposalShape => {
    return () => {
```
and we have to insert this line before the return in `sell`.
```
    const makeBidInvitation = buildBidInvitationMaker(BidProposalShape);
```